### PR TITLE
[CL-166] add title and subtitle inputs to bit-dialog

### PR DESCRIPTION
**Pull request by willmartian**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

- [CL-166] Add inputs `title` and `subtitle` to `bit-dialog`
- Update `bit-dialog` semantics to use html `section`, `header`, and `footer` elements (no visual changes)
- Update dialog title styles to match design spec (semi-bold font)

## Screenshots
image

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

[CL-166]: https://bitwarden.atlassian.net/browse/CL-166?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/components/src/dialog/dialog/dialog.stories.ts [hunk 1]
```diff
@@ -17,8 +17,14 @@ export default {
...
     moduleMetadata({
-      imports: [ButtonModule, SharedModule, IconButtonModule, TabsModule],
-      declarations: [DialogTitleContainerDirective, DialogCloseDirective],
+      imports: [
+        DialogModule,
+        ButtonModule,
+        SharedModule,
+        IconButtonModule,
+        TabsModule,
+        NoopAnimationsModule,
```
Comment (willmartian) [inline libs/components/src/dialog/dialog/dialog.stories.ts:26]: I added `NoopAnimationsModule` to this story to fix an unrelated runtime error I came across

File: libs/components/src/dialog/dialog/dialog.component.html [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
+  <header
     class="tw-flex tw-items-center tw-gap-4 tw-border-0 tw-border-b tw-border-solid tw-border-secondary-300 tw-p-4"
   >
-    <h1 bitDialogTitleContainer class="tw-mb-0 tw-grow tw-truncate tw-text-lg">
-      <ng-content select="[bitDialogTitle]"></ng-content>
-    </h1>
+    <span class="tw-inline-flex tw-gap-2 tw-grow tw-items-baseline tw-truncate">
+      <h1 bitDialogTitleContainer bitTypography="h4" noMargin class="tw-mb-0 tw-truncate">
+        {{ title }}
+        <ng-content select="[bitDialogTitle]"></ng-content>
```
Comment (willmartian) [inline libs/components/src/dialog/dialog/dialog.component.html:12]: My thinking here is to deprecate `[bitDialogTitle]` in favor of the `title` input.

Comment (Hinton) [reply libs/components/src/dialog/dialog/dialog.component.html:12]: Do you know if we currently use badges or something like that?

File: libs/components/src/dialog/dialog/dialog.component.html [hunk 3]
```diff
@@ -1,14 +1,20 @@
...
+      </h1>
+      <span *ngIf="subtitle" bitTypography="body2" class="tw-text-muted tw-truncate tw-flex-1">{{
+        subtitle
+      }}</span>
```
Comment (Hinton) [inline libs/components/src/dialog/dialog/dialog.component.html:16]:
```suggestion
      <span *ngIf="subtitle" bitTypography="body2" class="tw-text-muted tw-truncate tw-flex-1">
        {{ subtitle }}
      </span>
```

**Review by Hinton — COMMENTED** [review]
If we want to deprecate the title slot we should attempt to remove all documentation of it.

File: libs/components/src/dialog/dialog/dialog.component.html (hunk 2 above)
Comment (willmartian) [reply libs/components/src/dialog/dialog/dialog.component.html]: After a non-exhaustive search I only found the [member-dialog](https://github.com/bitwarden/clients/blob/ee4aa314443fb55d02d810825cec7fd186a71541/apps/web/src/app/admin-console/organizations/members/components/member-dialog/member-dialog.component.html#L8).

I think it is an option we want to keep open, and it was added to the design spec [here](https://www.figma.com/file/Zt3YSeb6E6lebAffrNLa0h?type=design%27&node-id=11910:18407).

**Review by Hinton — APPROVED** [review]
LGTM, let's create some tickets and put on teams backlog to get rid of the old behaviour. (Except for members dialog)

**Pull request merged**